### ISSUE TITLE
duckdb file size increases with CREATE & REPLACE

### ISSUE THREAD

**Issue by vinaychuri** [duckdb/duckdb#9429]
### What happens?

duckdb version '0.9.0' also tried with '0.9.1'
Python version 'Python 3.11.5'
MacOS 13.5.2 Ventura

```
df.shape
```
#(1166561, 30)

I save the dataframe to file via the following execute command.
```
with duckdb.connect(dbfile) as conn:
        conn.execute(f"CREATE OR REPLACE TABLE {df_name} AS SELECT * FROM {df_name}")

```
DB file size after the very first run is 211.3 MB
DB file size after re-run of same df is 422.9MB
DB file size after re-run of same df is 423.1MB
DB file size after re-run of same df is 422.9MB
DB file size after re-run of same df is 423.1MB
DB file size after re-run of same df is 423.6MB
Now create new .db file connection but same dataframe
New DB file size after the very first run is 211.3 MB

I should expect the db file size to remain constant at 211.3MB since the SQL command has a REPLACE
So how is the file size increasing with repetition?

I even tried to drop the table first before creation but still same results in file size.

```
with duckdb.connect(dbfile) as conn:
       conn.execute(f"DROP TABLE IF EXISTS my_table ")
        conn.execute(f"CREATE OR REPLACE TABLE {df_name} AS SELECT * FROM {df_name}")

```

### To Reproduce

```
dbfile=<path to database.db>'

with duckdb.connect(dbfile) as conn:
        conn.execute(f"CREATE OR REPLACE TABLE my_table AS SELECT * FROM {temp_df}")

```

### OS:

Mac OS Ventura

### DuckDB Version:

'0.9.0' also tried with '0.9.1'

### DuckDB Client:

Python 3.11.5

### Full Name:

Vinay Churi

### Affiliation:

None

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
This is expected behavior and is required to support this operation in an ACID manner. We first need to write the new table before we can delete the old table. What would perhaps be possible is vacuuming immediately afterwards but I would say that is orthogonal/already covered in other issues.

**Comment by vinaychuri** [comment]
Pls provide some more details on "vacuuming immediately afterwards". Can this be done via python code itself or needs to be done via CLI?
